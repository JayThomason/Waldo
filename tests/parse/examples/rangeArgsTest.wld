/** VERSION 2
 * checks that arguments to range are of correct type.
*/


PingPong

Endpoint Ping;
Endpoint Pong;


Sequences
{
}

Shared
{
}


/**** Ping endpoint *********/
Ping
{

    
    Public Function initiateSend()
    {
        List (element: Number) mList = range(0,10,-1);
        range(3+3,20-4,1);
        range(0,'a',1);
    }
}


Pong
{
}
