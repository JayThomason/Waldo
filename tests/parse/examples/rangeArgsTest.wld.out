*************************
Error in calling range.  Range requires all of its arguments to be numbers.  
Instead, the #2 argument you passed in to range was of type Text. 
-------
Line numbers:
30
30
-------
Program text:



27        {
28            List (element: Number) mList = range(0,10,-1);
29            range(3+3,20-4,1);
30*           range(0,'a',1);
31        }
32    }
33    
34    



27        {
28            List (element: Number) mList = range(0,10,-1);
29            range(3+3,20-4,1);
30*           range(0,'a',1);
31        }
32    }
33    
34    
*************************



